### DISCUSSION TITLE
Is it possible to disconnect a Remote

### DISCUSSION THREAD

**Discussion by Dennis1425** [pybricks/support#1445] [Q&A]
Using a Remote Control I would like to be able to disconnect it, so another hub can connect, without having to shutdown the first hub.
Or is it possible to restart a hub?

**Comment by laurensvalk** [comment]
If you just want to stop a program, you could call:

```python
raise SystemExit
```

This will also disconnect the remote.

There isn't currently a dedicated method to close the connection and keep the program running and start accepting another connection. However, with recent Bluetooth improvements this should be a reasonable possibility for implementing if there is a compelling use case for it. What is your use case?

**Reply by Dennis1425** [reply]
I want to control different hubs with the same remote, without having to touch the hubs, while also not using hub to hub connection.

Specifically I have the Lego liebherr crane 42146, it has 2 hubs.
One Hub for Driving and Rotating.
One for the crane stuff, boom, jib and hoist.

I dont really need both hubs at the same time, but switching with a button on the remote would be awesome.
I currently have implemented it by each hub searching for a specific remote name, and on pressing a button I would change the name to the one the other hub is searching for and disconnecting from the remote.

**Reply by laurensvalk** [reply]
> while also not using hub to hub

Why not?

**Comment by Dennis1425** [comment]
Well fair enough, the hubs arent ever gonna move away from each other, so I wouldnt need it in this case, although it still seems easier to me if i could switch hubs.

I also have a transporter, ideally I would like to turn both vehicles on once, and then be able to switch between them, using the same remote.

In this case I might drive one far enough away from the other, so that hub to hub wouldnt be realiable any more, however i have not tested the effective range.

I guess i could just put a hub on the remote, and then always use hub to hub, but i would like to avoid that just because the hub is heavy.

**Comment by Debenben** [comment]
Unfortunately I cannot test this at the moment, but if I think it might be possible using LWP3 device write commands (https://lego.github.io/lego-ble-wireless-protocol-docs/#action-types) like this

```python
from pybricks.pupdevices import Remote
from pybricks.tools import wait
my_remote = Remote()
wait(1000)

# my_remote.write(bytes([0x04, 0x00, 0x02, 0x01])) # shutdown
my_remote.write(bytes([0x04, 0x00, 0x02, 0x02])) # disconnect

wait(1000)
```

and possibly catching subsequent errors.

**Reply by Debenben** [reply]
Since I do not have my remote at the moment, I tested connecting and disconnecting a TechnicHub with official Lego firmware, I would expect this to work for the remote aswell. When trying this with the remote, change `LWP3Device(0x80)` to `LWP3Device(0x42)`

```python
from pybricks.iodevices import LWP3Device
from pybricks.tools import wait
my_remote = LWP3Device(0x80)


while(True):
    print("disconnect")
    my_remote.write(bytes([0x04, 0x00, 0x02, 0x02]))
    wait(2000)
    print("connect again")
    my_remote = LWP3Device(0x80)
    wait(2000)

**Reply by laurensvalk** [reply]
Very creative use to disconnect it from "the other end" :+1: :smile:

**Comment by laurensvalk** [comment]
See also https://github.com/pybricks/support/issues/802. I guess we should just add this feature :smile:

**Comment by vanczakp** [comment]
You can create a mode switch using button combination for example, then the top hub should be only connected to a remote (it can be xbox as well) and broadcast the required controls to the bottom hub. It is simple and works
